core/src/main/java/org/pentaho/di/connections/ConnectionManager.java

```diff
@@ -102,33 +135,41 @@ public synchronized void reset() {
   /**
    * This getter should not generally be used because it is limited to the global scope. It would be better to have
    * almost all callers use Bowl.getConnectionManager().
-   *
+   * <p>
    * This instance may still be used to register ConnectionProviders and Lookup Filters.
    *
    * @return ConnectionManager
    */
+  @NonNull
   public static ConnectionManager getInstance() {
+    // Lazy initialization is needed at least for being able to test the global connection manager with a custom,
+    // for initializing the global bowl's meta-store before this one is created.
+    if ( instance == null ) {
+      instance = new ConnectionManager();
+    }
+
     return instance;
   }
 
   /**
-   * Construct a new instance of a ConnectionManager using the metastore from the supplier.
-   *
-   * Instances returned by this will not share in-memory state with any other instannces. If you need the
+   * Construct a new instance of a ConnectionManager associated with a given bowl.
+   * <p>
+   * Instances returned by this use the metastore of the bowl.
+   * <p>
+   * Instances returned by this will not share in-memory state with any other instances. If you need the
    * ConnectionManager for a Bowl, use Bowl.getConnectionManager() instead.
    *
-   *
-   * @param bowl
-   *
+   * @param bowl The bowl.
    * @return ConnectionManager
    */
-  public static ConnectionManager getInstance( Supplier<IMetaStore> metastoreSupplier ) {
-    ConnectionManager newManager = new ConnectionManager();
-    newManager.setMetastoreSupplier( metastoreSupplier );
+  @NonNull
+  public static ConnectionManager getInstance( @NonNull Bowl bowl ) throws MetaStoreException {
```

Comment:
This does not work with the code currently on the project-profile branch, which has multiple metastores per Bowl (for RO and RW cases). We *might* be moving away from that, though.

Reply:
Latest version addresses this, putting back the meta-store supplier for additional control of which meta-store of the bowl to use.